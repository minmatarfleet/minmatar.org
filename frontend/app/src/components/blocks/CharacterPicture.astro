---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

import type { EvEImageServiceSize, ImageSize } from '@dtypes/layout_components'

interface Props {
    character_id:       number;
    character_name?:    string;
    x_character_id?:    string | false;
    x_character_name?:  string | false;
    size?:              ImageSize;
    icon_quality?:      EvEImageServiceSize;
    tooltip?:           string;
    mask?:              boolean;
    inert?:             boolean;
}

const {
    character_id,
    character_name,
    x_character_id = false,
    x_character_name = false,
    size = 64,
    icon_quality = 64,
    tooltip,
    mask = false,
    inert = false,
} = Astro.props

import { get_player_icon } from '@helpers/eve_image_server'
---

<picture
    class={inert ? 'animated' : undefined}
    data-tippy-content={tooltip ? tooltip : undefined}
    x-init={tooltip ? 'tippy($el, tippy_options)' : undefined}
>
    <img
        class:list={{ mask }}
        loading="lazy"
        height={size}
        width={size}
        src={get_player_icon(character_id, icon_quality ?? size)}
        x-bind:src={`${x_character_id} ? get_player_icon(${x_character_id}) : '${get_player_icon(character_id, icon_quality ?? size)}'`}
        alt={`${t('profile_picture_of')} ${character_name ?? t('unknown_character')}`}
        x-bind:alt={`\`${t('profile_picture_of')} \${(${x_character_name} ? ${x_character_name} : "${character_name ?? t('unknown_character')}")}\``}
        x-data={`{
            show_character_modal() {
                if (!${x_character_id} && ${character_id} == 0)
                    return

                show_modal({
                    partial: '${translatePath('/partials/character_modal_component/')}?character_id='+(${x_character_id} ? ${x_character_id} : '${character_id.toString()}'),
                })
            }
        }`}
        x-on:click.stop.prevent={inert ? undefined : 'show_character_modal()'}
    />
</picture>

<style lang="scss">
    img {
        transition: var(--fast-transition);
        filter: brightness(1);
        object-fit: cover;
        width: 100%;
    }

    picture:not(.animated) {
        img {
            cursor: pointer;
            
            &:hover {
                filter: brightness(1.3);
            }
        }
    }

    .mask {
        -webkit-mask-image: linear-gradient(to bottom,rgb(0,0,0) 0%,rgba(0,0,0,.7) 50%,transparent 100%);
        mask-image: linear-gradient(to bottom,rgb(0,0,0) 0%,rgba(0,0,0,.7) 50%,transparent 100%);
    }
</style>