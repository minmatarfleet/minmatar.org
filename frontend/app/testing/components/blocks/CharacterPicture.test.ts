import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CharacterPicture from "@components/blocks/CharacterPicture.astro";

import { get_player_icon } from "@helpers/eve_image_server";
vi.mock("@helpers/eve_image_server");

test("CharacterPicture defaults", async () => {
  vi.mocked(get_player_icon).mockImplementation((id, size) => {
    return "mock-picture-src";
  });

  const container = await AstroContainer.create();
  const result = await container.renderToString(CharacterPicture, {
    props: {
      character_id: 42,
      x_character_id: 765,
      character_name: "chicklet",
    },
  });

  expect(result).toMatchSnapshot();
});
