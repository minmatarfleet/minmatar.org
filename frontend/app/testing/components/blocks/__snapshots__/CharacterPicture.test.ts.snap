// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterPicture defaults 1`] = `
"<picture data-astro-cid-hwwa7zih> <img loading="lazy" height="64" width="64" src="mock-picture-src" x-bind:src="765 ? get_player_icon(765) : 'mock-picture-src'" alt="Profile picture of chicklet" x-bind:alt="\`Profile picture of \${(false ? false : &#34;chicklet&#34;)}\`" x-data="{
            show_character_modal() {
                if (!765 &#38;&#38; 42 == 0)
                    return

                show_modal({
                    partial: '/partials/character_modal_component/?character_id='+(765 ? 765 : '42'),
                })
            }
        }" x-on:click.stop.prevent="show_character_modal()" data-astro-cid-hwwa7zih> </picture> "
`;
